### ISSUE TITLE
Flatten results

### ISSUE THREAD

**Issue by plastbox** [reminyborg/aggsy#10]
**What:**
The ability to flatten results from Aggsy, using bucket names as property names. Based on README.md's examples:
```javascript
var aggsy = require('aggsy')

var cars = [
  { model: 'volvo', make: 'v50', km: 100 },
  { model: 'tesla', make: 's', km: 200 },
  { model: 'tesla', make: 's', km: 120 },
  { model: 'tesla', make: 'x', km: 10 }
]

aggsy('model(distance: _sum(km), reports: _count())', cars)

// Gives:
{
  tesla: { 'reports': 2, 'distance': 330 },
  volvo: { 'reports': 1, 'distance': 100 }
}

aggsy('model(distance: _sum(km), reports: _count())', cars, {flatten: true})

// Could give:
[
  { 'model': 'tesla', 'reports': 2, 'distance': 330 },
  { 'model': 'volvo', 'reports': 1, 'distance': 100 }
]
```

Nested groups requires handling of name collitions. One solution could be to introduce an option to force path prefixing:
```javascript
aggsy('model(make(count: _count()), count: _count())', cars)

// Gives:
{
  tesla: {
    's': { 'count': 2 },
    'x': { 'count': 1 },
    'count': 3
  },
  volvo: {
    'v50': { 'count': 1 },
    'count': 1
  }
}

aggsy('model(make(count: _count()), count: _count())', cars, {flatten: true})
// Throws error due to collition between properties named _count_.

aggsy('model(make(count: _count()), count: _count())', cars, {flatten: true, prefix: true})
// Could give:
[
  {model: 'tesla', 'model.make': 's', 'model.make.count': 2, 'model.count': 3},
  {model: 'tesla', 'model.make': 'x', 'model.make.count': 2, 'model.count': 3},
  {model: 'volvo', 'model.make': 'v50', 'model.make.count': 1 'model.count': 1},
]

// Optionally pick non-conflicting names:
aggsy('model(make(count: _count()), count_total: _count())', cars, {flatten: true})
// Could give:
[
  {model: 'tesla', 'make': 's', 'count': 2, 'count_total': 3},
  {model: 'tesla', 'make': 'x', 'count': 2, 'count_total': 3},
  {model: 'volvo', 'make': 'v50', 'count': 1 'count_total': 1},
]

```

**Why**
The convenience of getting data out of a tool as flexible as Aggsy in a more uniform format.

Easier to feed into other components or data processes, or indeed into a new Aggsy instance.

Easier conversion to csv.

**Comment by reminyborg** [comment]
Got ready to flatten by doing https://github.com/reminyborg/aggsy/commit/55fc23f55149a868b46bb2b8481be652d8e1ff2b and https://github.com/reminyborg/aggsy/commit/777077775e3fbaa08b6cd8003a5c895fb4bc88b7

**Comment by reminyborg** [comment]
This is sorted out by 310bdf9cd4ad7061cddb7695b45a2bb67ad4974a and cfd3e8239aa7d94bb907fb440d17c0580f8d1510.

Released with version 1.3.0.

I would not limit the duplication of naming when we flatten groups.

As we flatten return a flat result. If there is duplicate named reducers in the nested groups, we can actually use this to reduce multiple values from different parts of the tree into one value.

We might want to warn against this in the documentation tho.
